lib/logstash/outputs/kafka.rb

```diff
@@ -188,22 +190,20 @@ def register
 
 
     @producer = create_producer
-    @codec.on_event do |event, data|
-      begin
-        if @message_key.nil?
-          record = org.apache.kafka.clients.producer.ProducerRecord.new(event.sprintf(@topic_id), data)
-        else
-          record = org.apache.kafka.clients.producer.ProducerRecord.new(event.sprintf(@topic_id), event.sprintf(@message_key), data)
-        end
-        prepare(record)
-      rescue LogStash::ShutdownSignal
-        @logger.debug('Kafka producer got shutdown signal')
-      rescue => e
-        @logger.warn('kafka producer threw exception, restarting',
-                     :exception => e)
+    if value_serializer == 'org.apache.kafka.common.serialization.StringSerializer'
+      @codec.on_event do |event, data|
+        write_to_kafka(event, data)
```

Comment:
I suggest doing a `write_to_kafka(event, serialize_data(data))` so we can remove the duplicate code.

Reply:
@jsvd with `serialize_data` as some dynamic method set up accordingly (otherwise we have to evaluate the conditional for every event)?

Reply:
fair enough, let's leave as-is.